src/Sylius/Component/Core/Dashboard/SalesDataProvider.php

```diff
@@ -24,40 +26,37 @@ final class SalesDataProvider implements SalesDataProviderInterface
     /** @var EntityManagerInterface */
     private $entityManager;
 
-    public function __construct(EntityManagerInterface $entityManager)
+    /** @var ServiceLocator */
+    private $providerLocator;
+
+    public function __construct(EntityManagerInterface $entityManager, ServiceLocator $providerLocator)
     {
         $this->entityManager = $entityManager;
+        $this->providerLocator = $providerLocator;
     }
 
     public function getLastYearSalesSummary(ChannelInterface $channel): SalesSummaryInterface
     {
-        $startDate = (new \DateTime('first day of next month last year'))->format('Y/m/d');
-        $endDate = (new \DateTime('last day of this month'))->format('Y/m/d');
-        $channelId = $channel->getId();
-
-        $query = $this->entityManager->getConnection()->query(
-            "SELECT
-                DATE_FORMAT(checkout_completed_at, '%m.%y') AS \"date\",
-                SUM(total) as \"total\"
-            FROM sylius_order
-            WHERE (channel_id = $channelId)
-            AND (checkout_completed_at BETWEEN '$startDate' AND '$endDate')
-            AND (payment_state = 'paid')
-            GROUP BY date;"
-        );
+        $startDate = (new \DateTime('first day of next month last year'));
+        $startDate->setTime(0, 0, 0);
+        $endDate = (new \DateTime('last day of this month'));
+        $endDate->setTime(23, 59, 59);
 
-        $query->execute();
-        $result = $query->fetchAll();
-
-        $data = [];
-        foreach ($result as $item) {
-            $data[$item['date']] = (int) $item['total'];
-        }
+        $data = $this->getLastYearSalesProvider()->getSalesSummary($channel, $startDate, $endDate);
 
         return new SalesSummary(
-            (new \DateTime('first day of next month last year')),
-            (new \DateTime('last day of this month')),
+            $startDate,
+            $endDate,
             $data
         );
     }
+
+    private function getLastYearSalesProvider(): LastYearSalesSummaryProviderInterface
```

Comment:
It feels like this private method is breaking Open/Closed principle. Let's say I want to add a dedicated `LastYearSalesSummaryProviderInterface` instance - it'd require me to overwrite any function like this in any place `LastYearSalesSummaryProviderInterface` is used in.

This should be an internal implementation detail of `LastYearSalesSummaryProviderInterface` - I shouldn't be even aware of it while using it, my application code shouldn't be concerned with infrastructural stuff.